simpleflow/executor.py

```diff
@@ -42,23 +44,30 @@ def __init__(self, workflow):
         as a program, the workflow, and an interpreter, the executor.
 
         """
-        self._workflow = workflow(self)
+        self._workflow_class = workflow_class
+        self._workflow = None
+
+    @property
+    def workflow_class(self):
+        return self._workflow_class
 
     @property
     def workflow(self):
-        """
-        :return:
-        :rtype: simpleflow.workflow.Workflow
-        """
         return self._workflow
 
+    def create_workflow(self):
+        if self._workflow is None:
+            workflow = self._workflow_class(self)
+            if False:
+                assert isinstance(workflow, Workflow)
```

Comment:
Ah! Leave it then. I understood the imports and additional comments, but this one was more mysterious ;-)